### ISSUE TITLE
Play Store now requires minimum SDK version 29

### ISSUE THREAD

**Issue by mendhak** [mendhak/gpslogger#784]
[From November 1, the SDK needs to be at v 29.](https://support.google.com/googleplay/android-developer/answer/113469#targetsdk)

When upgrading to targetSdkVersion=29, some of the interfaces and methods are deprecated and no longer work with SDK version less than 24.  

This excludes a lot of Android 4, 5, 6 devices.  

Try to find a way to allow them to coexist.

**Comment by mendhak** [comment]
class GeneralLocationListener implements LocationListener, GpsStatus.Listener, GpsStatus.NmeaListener, android.location.OnNmeaMessageListener {

The last bit, `android.location.OnNmeaMessageListener` requires API 24. Using it on older Android versions results in

> E/Event: Could not dispatch event: class com.mendhak.gpslogger.common.events.CommandEvents$RequestToggle to subscribing class class com.mendhak.gpslogger.GpsLoggingService
    java.lang.NoClassDefFoundError: com.mendhak.gpslogger.GeneralLocationListener

**Comment by mendhak** [comment]
**NMEA Listener**

After a lot of trying and digging - I've found that Google [accidentally removed an API](https://issuetracker.google.com/issues/141019880) instead of deprecating it.  So the code I wrote to get NMEA listener working with the newer SDK means it won't work at all for older devices.   It will be reinstated in SDK 30.  

There is a [workaround here](https://stackoverflow.com/questions/57975969/accessing-nmea-on-android-api-level-24-when-compiled-for-target-api-level-29) involving reflection and at first glance the complexity is increasing.  This isn't a great situation - it'll be extremely error prone, and what effect does it have on stability?   How to test this? 

And even though this is a temporary fix, there's nothing more permanent than something temporary.  This will hang around for a long time.
